src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,135 @@
+                                         namespace="",
+                                         kind="",
+                                         package="unknown",
+                                         args="unknown",
+                                         remappings={},
+                                         launch_filename="unknown",
+                                         app=self._app_instance,
+                                         files=self._app_instance.files,
+                                         params=ParameterServer(),
+                                     ))
+                                    for node in nodes)
+                # Places to store bits of actions, which only appear as topics
+                action_server_candidates: Dict[str, Dict[str, ActionCandidate]] = dict()
+                action_client_candidates: Dict[str, Dict[str, ActionCandidate]] = dict()
+                # Only include topics that we care about
+                significant_publishers = ROS1Observer._filter_out_topics_and_nodes(info.publishers)
+                significant_subscribers = ROS1Observer._filter_out_topics_and_nodes(
+                    info.subscribers)
+
+                # Process the topics
+                separate_topics_from_action(nodecontexts,
+                                            significant_publishers, True,
+                                            action_client_candidates,
+                                            action_server_candidates, ros)
+                separate_topics_from_action(nodecontexts,
+                                            significant_subscribers, False,
+                                            action_client_candidates,
+                                            action_server_candidates, ros)
+
+                # Process the services
+                for service, nodes in info.services.items():
+                    if service.split('/')[-1] not in _SERVICES_TO_FILTER_OUT:
+                        for node in nodes:
+                            nodecontexts[node].provide(service,
+                                                       ros.services[service].format.fullname)
+
+                # Check if action candidates are complete (i.e., have all their topics)
+                # and add action if they are, or add the topics back in if they're not
+                self.process_action_candidates(action_server_candidates,
+                                               lambda node: nodecontexts[node].action_server,
+                                               nodecontexts)
+                self.process_action_candidates(action_client_candidates,
+                                               lambda node: nodecontexts[node].action_client,
+                                               nodecontexts)
+
+            self._nodes = nodecontexts
+            return self.summarise()
+        except TimeoutError:
```

Comment:
I'm not sure that this `try/except` is necessary since the exception _should_ contain an error message with roughly the same description. Removing the `try/except` would further help to improve readability.